### PR TITLE
meter shows "finished trips vs baseline" instead of time

### PR DESCRIPTION
**Pull request by michaelkirk**
Screen Shot 2021-03-15 at 4 34 47 PM

Screen Shot 2021-03-15 at 4 34 54 PM

https://user-images.githubusercontent.com/217057/111234586-177eff00-85ac-11eb-9238-59c46fb11e4e.mp4

## edge cases

**next day**
Screen Shot 2021-03-15 at 4 25 02 PM
**no prebaked day/night**
Screen Shot 2021-03-15 at 4 24 24 PM
Screen Shot 2021-03-15 at 4 24 17 PM

**no changes**
Screen Shot 2021-03-15 at 4 35 52 PM

### CODE REVIEW

File: game/src/sandbox/speed.rs [hunk 1]
```diff
@@ -156,34 +159,132 @@ impl TimePanel {
...
+            finished as f64 / total as f64
+        } else {
+            0.0
+        };
+        let finished_width = ratio * bar_width;
+
+        if app.has_prebaked().is_some() {
+            let now = self.time;
+            let mut baseline_finished = self.baseline_finished_trips.unwrap_or(0);
+            for (t, _, _, _) in &app.prebaked().finished_trips[baseline_finished..] {
```
Comment (dabreegster) [inline game/src/sandbox/speed.rs:184]: Ahhh, clever trick. Most of the things that read `Analytics` recalculate everything from scratch constantly. To one day speed this up, I was imagining making some kind of stateful "cursor" objects to remember where a computation left off. But that's total overkill for something like this -- just remembering the count is much simpler.

File: game/src/sandbox/speed.rs [hunk 2]
```diff
@@ -156,34 +159,132 @@ impl TimePanel {
...
+                if baseline_finished_width > 0.0 {
+                    let baseline_cursor = Polygon::rectangle(cursor_width, bar_height)
+                        .translate(baseline_finished_width, 0.0);
+                    progress_bar.push(cursor_fg, baseline_cursor);
+                }
+            }
+        }
+
+        let text_geom = Text::from(
+            Line(format!("Finished Trips: {}", prettyprint_usize(finished),)).fg(text_color),
```
Comment (dabreegster) [inline game/src/sandbox/speed.rs:228]: nit: stray comma inside the format macro

File: game/src/sandbox/speed.rs [hunk 3]
```diff
@@ -156,34 +159,132 @@ impl TimePanel {
...
+            );
+
+            progress_bar.push(cursor_fg, triangle_poly);
+        }
+
+        use widgetry::DrawWithTooltips;
+        let mut tooltip_text = Text::from(Line("Finished Trips"));
+        tooltip_text.add(Line(format!(
+            "{} ({}% of total)",
+            finished,
```
Comment (dabreegster) [inline game/src/sandbox/speed.rs:260]: `prettyprint_usize(finished)`

File: game/src/sandbox/speed.rs [hunk 4]
```diff
@@ -156,34 +159,132 @@ impl TimePanel {
...
+            (ratio * 100.0) as usize
+        )));
+        if let Some(baseline_finished) = self.baseline_finished_trips {
+            // TODO: up/down icons
+            let line = if baseline_finished > finished {
+                let difference = baseline_finished - finished;
+                Line(format!("{} less than baseline", difference))
```
Comment (dabreegster) [inline game/src/sandbox/speed.rs:267]: `prettyprint_usize` here and below, to help point out really dramatically high differences from the baseline

**Review by dabreegster — CHANGES_REQUESTED** [review]
Awesome job! I think this strikes a great balance between easy to learn the meaning of (with the tooltip) and succinct to glance at after knowing.

**Comment by michaelkirk** [comment]
Ok, I believe I've responded to your feedback. PTAL @dabreegster

**Comment by dabreegster** [comment]
Thanks for the PR!

**Pull request merged**